utility/install_prereq.py

```diff
@@ -49,77 +93,44 @@ def run(nodes):
-    """Enables tools repositories."""
-
-    @staticmethod
-    def run(nodes):
-        nodes = nodes if isinstance(nodes, list) else [nodes]
-        for node in nodes:
-            stdout, _ = os_major_version(node)
-            repos = [RHCEPH5_TOOLS_REPO.format(stdout.strip())]
-            if stdout.strip() == "8":
-                repos.append(RHEL8_ANSIBLE_REPO)
-
-            if SubscriptionManager(node).repos.enable(repos):
-                raise EnableToolsRepositoriesError("Failed to enable tools repo.")
-
-        log.info("Enabled tools repos on cluster nodes successfully.")
-
-
-class ConfigureCephadmRepositories:
+class ConfigureCephToolsRepositories:
     """Configures Ceph and Cephadm related repositories."""
 
     @staticmethod
-    def run(nodes, args, config):
+    def run(nodes, build_type, base_url, rhbuild):
         """Install and configure cephadm package.
 
         Args:
-            args (dict): key/value pairs passed from the test case
-            config (dict): key/value pairs passed from the cluster config
+            nodes (CephNode | List ): CephNode or list of CephNode object
+            build_type (str): Build type
+            base_url (str): Custom repo build url
+            rhbuild (str): RHCS build version with RHEL
         """
-        nodes = nodes if isinstance(nodes, list) else [nodes]
-        pkg = Package(nodes)
-
-        custom_repo = args.get("custom_repo")
-        rhcs_version = args.get("rhcs-version")
-        rhcs_release = args.get("release")
-
-        build_type = config.get("build_type")
-        rhbuild = config.get("rhbuild")
-        base_url = config.get("base_url")
-        cloud_type = config.get("cloud-type", "openstack")
-
-        if rhcs_release and rhcs_version:
-            _platform = "-".join(rhbuild.split("-")[1:])
-            _details = fetch_build_artifacts(rhcs_release, rhcs_version, _platform)
-            config["base_url"] = _details[0]
-
-        if build_type == "upstream":
-            pkg.add_repo(config.get("base_url"))
-            pkg.install(RHEL8_EPEL_RELESE_RPM)
-            pkg.clean()
-
-            # Enabling cdn tool repo to workaround, due to unavailability of
-            # ceph x86_64 RPM pkgs in upstream builds
-            EnableToolsRepositories.run(nodes)
-        elif build_type == "released" or custom_repo.lower() == "cdn":
-            EnableToolsRepositories.run(nodes)
-        elif custom_repo:
-            base_url = get_custom_repo_url(repo=custom_repo)
-            pkg.add_repo(base_url, nogpgcheck=True)
-            pkg.clean()
-
-            log.info(f"Added repo {base_url} on cluster nodes successfully.")
+        if build_type in ["latest", "tier-0", "tier-1", "tier-2", "rc", "upstream"]:
+            Package(nodes).add_repo(base_url)
+
+        elif build_type in ["ga", "ga-async"]:
+            rhcs, rhel = get_builds_by_rhbuild(rhbuild)
+            repos = RHCS5_REPOS
+            if rhcs == 6:
```

Comment:
if possible use switch case or getattr(by defining class for each release) or release config yaml files  for getting repos, which will be used as generic way to fetch things to enable.